### PR TITLE
Implement `--checksum` option on `swift sdk install`

### PR DESCRIPTION
**Pull request by MaxDesiatov**
### Motivation:

This option [was specified in the corresponding proposal for Swift SDKs](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0387-cross-compilation-destinations.md#swift-sdk-installation-and-configuration):

> For Swift SDKs installed from remote URLs an additional `--checksum` option is required, through which users of a Swift SDK can specify a checksum provided by a publisher of the SDK. The latter can produce a checksum by running `swift package compute-checksum` command (introduced in [SE-0272](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0272-swiftpm-binary-dependencies.md)) with the Swift SDK bundle archive as an argument.

### Modifications:

Added `isFileSupported` method on `Archiver` to unify archive extensions matching logic between `swift package compute-checksum` and `swift sdk install`.

Refactored checksum computation logic into `static func checksum` on `Workspace.BinaryArtifactsManager` to also unify it between the two commands.

Added error and output handling for checksums in `func SwiftSDKBundleStore.installIfValid`, updated corresponding unit tests.

### Result:

Resolves rdar://130590711

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by kateinoigakukun** [comment]
@swift-ci test windows

**Review by kateinoigakukun — APPROVED** [review]

File: Sources/PackageModel/SwiftSDKs/SwiftSDK.swift [hunk 1]
```diff
@@ -64,6 +70,17 @@ public enum SwiftSDKError: Swift.Error {
...
         switch self {
+        case let .checksumInvalid(computed, provided):
+            return """
+            Computed archive checksum `\(computed) does not match the provided checksum `\(provided)`.
+            """
+
+        case .checksumNotProvided(let url):
+            return """
+            Bundles installed from remote URLs (such as \(url)) require their checksum passed via `--checksum` option.
+            The distributor of the bundle must compute it with `swift package compute-checksum` \
```
Comment (al45tair) [inline Sources/PackageModel/SwiftSDKs/SwiftSDK.swift:81]:
```suggestion
            The distributor of the bundle must compute it with the `swift package compute-checksum` \
```

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Review by xedin — APPROVED** [review]

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
CI failure looks unrelated:
```
/Users/ec2-user/jenkins/workspace/swift-package-manager-PR-macos-smoke-test/branch-main/sourcekit-lsp/Sources/SourceKitLSP/Rename.swift:625:40: error: type 'Symbol' does not conform to the 'Sendable' protocol
```

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Pull request merged**